src/transformers/models/superpoint/modeling_superpoint.py

```diff
@@ -0,0 +1,577 @@
+        return_dict: Optional[bool] = True,
+    ) -> Union[Tuple, BaseModelOutputWithNoAttention]:
+        all_hidden_states = () if output_hidden_states else None
+        input = self.relu(self.conv1a(input))
+        input = self.relu(self.conv1b(input))
+        input = self.pool(input)
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (input,)
+
+        input = self.relu(self.conv2a(input))
+        input = self.relu(self.conv2b(input))
+        input = self.pool(input)
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (input,)
+
+        input = self.relu(self.conv3a(input))
+        input = self.relu(self.conv3b(input))
+        input = self.pool(input)
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (input,)
+
+        input = self.relu(self.conv4a(input))
+        output = self.relu(self.conv4b(input))
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (output,)
+
+        if not return_dict:
+            return tuple(v for v in [output, all_hidden_states] if v is not None)
+
+        return BaseModelOutputWithNoAttention(
+            last_hidden_state=output,
+            hidden_states=all_hidden_states,
+        )
+
+
+class SuperPointInterestPointDecoder(nn.Module):
+    """
+    The SuperPointInterestPointDecoder uses the output of the SuperPointEncoder to compute the keypoint with scores.
+    The scores are first computed by a convolutional layer, then a softmax is applied to get a probability distribution
+    over the 65 possible keypoint classes. The keypoints are then extracted from the scores by thresholding and
+    non-maximum suppression. Post-processing is then applied to remove keypoints too close to the image borders as well
+    as to keep only the k keypoints with highest score.
+    """
+
+    def __init__(self, config: SuperPointConfig) -> None:
+        super().__init__()
+        self.hidden_sizes = config.hidden_sizes
+        self.descriptor_dim = config.descriptor_dim
+        self.keypoint_threshold = config.keypoint_threshold
+        self.max_keypoints = config.max_keypoints
+        self.nms_radius = config.nms_radius
+        self.border_removal_distance = config.border_removal_distance
+
+        self.relu = nn.ReLU(inplace=True)
+        self.pool = nn.MaxPool2d(kernel_size=2, stride=2)
+        self.convSa = nn.Conv2d(
+            self.hidden_sizes[3],
+            self.hidden_sizes[4],
+            kernel_size=3,
+            stride=1,
+            padding=1,
+        )
+        self.convSb = nn.Conv2d(self.hidden_sizes[4], 65, kernel_size=1, stride=1, padding=0)
+
+    def forward(self, encoded: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        # Compute the dense keypoint scores
+        scores = self.get_scores(encoded)
+        # Extract keypoints
+        keypoints, scores = self.extract_keypoints(scores)
+
```

Comment:
Comments are necessary - the methods are well-named so it's clear what's happening :) 

```suggestion
        scores = self.get_scores(encoded)
        keypoints, scores = self.extract_keypoints(scores)
```

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/25786/commits/c43052de41132765b2bdfccb863a0ee7aa33a9c4)